### PR TITLE
[SPARK-36848][SQL] Migrate ShowCurrentNamespaceStatement to v2 command framework

### PR DESCRIPTION
**Pull request by huaxingao**
### What changes were proposed in this pull request?

Migrate `ShowCurrentNamespaceStatement` to v2 command framework

### Why are the changes needed?
Migrate to the standard V2 framework

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Existing tests

### CODE REVIEW

**Comment by huaxingao** [comment]
cc @cloud-fan

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala [hunk 1]
```diff
@@ -3566,11 +3566,11 @@ class AstBuilder extends SqlBaseBaseVisitor[AnyRef] with SQLConfHelper with Logg
   }
 
   /**
-   * Create a [[ShowCurrentNamespaceStatement]].
+   * Create a [[ShowCurrentNamespace]].
    */
   override def visitShowCurrentNamespace(
       ctx: ShowCurrentNamespaceContext) : LogicalPlan = withOrigin(ctx) {
-    ShowCurrentNamespaceStatement()
+    ShowCurrentNamespace(None)
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala:3573]: This command does not need any analysis and using `RunnableCommand` should be good enough
```
case class ShowCurrentNamespaceCommand extends LeafRunnableCommand {
  override def run(sparkSession: SparkSession): Seq[Row] = {
    val catalogManager = sparkSession.sessionState.catalogManager
    ...
  }
}
```

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Strategy.scala [hunk 2]
```diff
@@ -330,7 +330,7 @@ class DataSourceV2Strategy(session: SparkSession) extends Strategy with Predicat
       SetCatalogAndNamespaceExec(catalogManager, catalogName, ns) :: Nil
 
     case r: ShowCurrentNamespace =>
-      ShowCurrentNamespaceExec(r.output, r.catalogManager) :: Nil
+      ShowCurrentNamespaceExec(r.output, r.catalogManager.get) :: Nil
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Strategy.scala:333]: We don't need to pass the catalog manager from logical plan to physical plan, we can get it directly in the physical plan
```
val catalogManager = session.sessionState.catalogManager
...
```
see  https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlan.scala#L60

Comment (cloud-fan) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Strategy.scala:333]: then the logical plan can be `case class ShowCurrentNamespace extends LeafCommand`

**Review by cloud-fan — APPROVED** [review]

**Review by viirya — APPROVED** [review]

**Comment by viirya** [comment]
Thanks! Merging to master.

**Comment by huaxingao** [comment]
Thanks!

**Pull request closed without merging**